.github/workflows/nightly-upstream-snapshot-build.yml

```diff
@@ -45,22 +45,6 @@ jobs:
         with:
           registry: public.ecr.aws
 
-      # cache local patch outputs
-      - name: Cache local Maven repository
-        id: cache-local-maven-repo
-        uses: actions/cache@v3
-        with:
-          path: |
-            ~/.m2/repository/io/opentelemetry/
-          key: ${{ runner.os }}-maven-local-${{ hashFiles('.github/patches/opentelemetry-java*.patch') }}
-
-      - name: Publish patched dependencies to maven local
```

Comment:
the reason for removing is that patching is not required in the nightly build workflows as it testing against upstream snapshot?